os/os.go

```diff
@@ -0,0 +1,159 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package os provides gNOI os operations.
+package os
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	log "github.com/golang/glog"
+	ospb "github.com/openconfig/gnoi/os"
+	"github.com/openconfig/gnoigo/internal"
+)
+
+// InstallOperation represents the parameters of a Install operation.
+type InstallOperation struct {
+	version string
+	standby bool
+	reader  io.Reader
+}
+
+// Version identifies the OS version.
+func (i *InstallOperation) Version(version string) *InstallOperation {
+	i.version = version
+	return i
+}
+
+// Standby specifies if supervisor is on standby.
+func (i *InstallOperation) Standby(standby bool) *InstallOperation {
+	i.standby = standby
+	return i
+}
+
+// Reader specifies the package reader for the OS file.
+func (i *InstallOperation) Reader(reader io.Reader) *InstallOperation {
+	i.reader = reader
+	return i
+}
+
+// NewInstallOperation creates an empty InstallOperation.
+func NewInstallOperation() *InstallOperation {
+	return &InstallOperation{}
+}
+
+// awaitPackageInstall receives messages from the client until either
+// (a) the package is installed and validated, in which case it returns the InstallResponse message
+// (b) the device does not have the package, in which case it returns a nil response
+// (c) an error occurs, in which case it returns the error
+func awaitPackageInstall(ic ospb.OS_InstallClient) (*ospb.InstallResponse, error) {
+	for {
+		cresp, err := ic.Recv()
+		if err != nil {
+			return nil, err
+		}
+		switch v := cresp.GetResponse().(type) {
+		case *ospb.InstallResponse_Validated:
+			return cresp, nil
+		case *ospb.InstallResponse_TransferReady:
+			return nil, nil
+		case *ospb.InstallResponse_InstallError:
+			errName := ospb.InstallError_Type_name[int32(v.InstallError.Type)]
+			return nil, fmt.Errorf("installation error %q: %s", errName, v.InstallError.GetDetail())
+		case *ospb.InstallResponse_TransferProgress:
+			log.Infof("installation progress: %v bytes received from client", v.TransferProgress.GetBytesReceived())
+		case *ospb.InstallResponse_SyncProgress:
+			log.Infof("installation progress: %v%% synced from supervisor", v.SyncProgress.GetPercentageTransferred())
+		default:
+			return nil, fmt.Errorf("unexpected client install response: %v (%T)", v, v)
+		}
+	}
+}
+
+func transferContent(ic ospb.OS_InstallClient, reader io.Reader) error {
+	// The gNOI SetPackage operation sets the maximum chunk size at 64K,
+	// so assuming the install operation allows for up to the same size.
+	buf := make([]byte, 64*1024)
+	for {
+		n, err := reader.Read(buf)
+		if n > 0 {
+			if err := ic.Send(&ospb.InstallRequest{
+				Request: &ospb.InstallRequest_TransferContent{
+					TransferContent: buf[0:n],
+				}}); err != nil {
```

Comment:
indenting here doesn't conform to go style: closing brace line should be at same indenting level as opening brace line

Reply:
done